src/draw/lv_image_decoder.h

```diff
@@ -76,6 +76,21 @@ typedef lv_result_t (*lv_image_decoder_get_area_cb_t)(lv_image_decoder_t * decod
  */
 typedef void (*lv_image_decoder_close_f_t)(lv_image_decoder_t * decoder, lv_image_decoder_dsc_t * dsc);
 
+#if LV_USE_MATRIX
+/**
+ * Custom drawing functions for special image formats.
+ * @param layer pointer to a layer
+ * @param dsc pointer to decoder descriptor
+ * @param coords the coordinates of the image
+ * @param matrix the transform of the image if `LV_USE_MATRIX` is enabled
+ */
+typedef void (*lv_image_decoder_custom_draw_t)(lv_layer_t * layer, const lv_image_decoder_dsc_t * dsc,
+                                               const lv_area_t * coords, const lv_matrix_t * matrix);
```

Comment:
Can we merge the parameters into a structure? This will make it easier to expand the parameters later.

Reply:
use `lv_draw_image_dsc_t` pointer instead